File: Problem/Easy/Problem_7/Solution.cpp
// C++ program using to recursive function to print the reverse of a given string.

#include <iostream>
#include <string>
#include <cstring>

// Recursive function
void printReverse(std::string s)
{
    if (s.length() == 0)
        return;
    printReverse(s.substr(1));
    std::cout << s[0];
}

void printConstChar(const char * str, int n)
{
    if (n == 0)
        return;
    std::cout << str[n - 1];
    printConstChar(str, n - 1);
}

// Main function
int main()
{
    // Solution 1
    std::string s = "Nhan Pham";
    printReverse(s);
    
    std::cout << std::endl << std::endl;
    
    // Solution 2
    const char * str = "Nhan Pham";
    printConstChar(str, strlen(str));
    
    return 0;
}

File: Problem/Easy/Problem_7/Solution.py
# Python3 program using recursive function to print the reverse of a given string.

# Recursive function
def reverseString(s : str) -> None:
    if len(s) == 0:
        return
    print(s[-1], end='')
    reverseString(s[:-1])

def reverseString_2(s : str) -> None:
    if len(s) == 0:
        return
    reverseString_2(s[1:])
    print(s[0], end='')

# Main
if __name__ == '__main__':
    s = "Nhan Pham"
    
    # Solution 1
    reverseString(s)
    
    print('')
    
    # Solution 2
    reverseString_2(s)